### PR TITLE
refactor: decommission `@nhost/core`

### PR DESCRIPTION
**Pull request by plmercereau**
third attempt
- https://github.com/nhost/nhost/pull/1108
- https://github.com/nhost/nhost/pull/1246

### CODE REVIEW

File: packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts
Comment (elitan) [inline packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts:66]: Why are there so many changes to `docgen/src/*`? It does not seem related to `core`.

**Review by elitan — COMMENTED** [review]
Looks good, but massive PR, so hard to go deep into every bit. I'm trusting the test.

File: packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts
Comment (plmercereau) [reply packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts:66]: I don't know, but as long as the reference documentation works that's fine by me.
You can ask Szilard if it bugs you

**Comment by plmercereau** [comment]
> Looks good, but massive PR, so hard to go deep into every bit. I'm trusting the test.

I will wait for Nuno's feedback before merging it.

File: packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts
Comment (elitan) [reply packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts:66]: ok

File: packages/hasura-storage-js/src/utils/types.ts
Comment (nunopato) [inline packages/hasura-storage-js/src/utils/types.ts:3]: :+1:

**Review by nunopato — APPROVED** [review]
looks good to me. @szilarddoro please have a look as well.

File: packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts
Comment (szilarddoro) [reply packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts:66]: I think you know what this was for @plmercereau, because you raised the issue to me, for which the fix was created. 😄

Things were breaking in DocGen after moving things from `core` to `hasura-auth-js` and `hasura-storage-js`. Some type references were unavailable with the previous setup, but they started working when we moved things. These type references were not properly covered by the auto-generator, so I had change a few things under the hood.

Comment (plmercereau) [reply packages/docgen/src/fragments/FunctionSignatureTypeFragment.ts:66]: right, I did not follow this was related. Everything works, so that's fine

**Pull request merged**
